content/courses/Guidelines_for_Safe_and_Secure_Ada_SPARK/chapters/guidelines/dynamic_storage_management/dyn03_access_types_without_allocators_policy.rst

```diff
@@ -0,0 +1,93 @@
+------------------------------------------------
+Access Types Without Allocators Policy (DYN03)
+------------------------------------------------
+
+**Level** :math:`\rightarrow` Advisory
+
+Category
+   :Safety: :math:`\checkmark`
+   :Cyber: :math:`\checkmark`
+
+Goal
+   :Maintainability: :math:`\checkmark`
+   :Reliability: :math:`\checkmark`
+   :Portability:
+   :Performance:
+   :Security: :math:`\checkmark`
+
+**Remediation** :math:`\rightarrow` Low
+
+**Verification Method** :math:`\rightarrow` Compiler restrictions
+
++++++++++++
+Reference
++++++++++++
+
+MISRA C Rule 21.3 "The memory allocation and deallocation functions of <stdlib.h>
+shall not be used"
+
++++++++++++++
+Description
++++++++++++++
+
+The following restrictions must be in effect:
+
+   * No_Allocators
+   * No_Dependence => Ada.Unchecked_Deallocation
+
+In this approach dynamic access values are only created via the attribute
+:ada:`'Access` applied to aliased objects. Allocation and deallocation never
+occur. As a result, storage exhaustion cannot occur because no *dynamic*
+allocations occur. Fragmentation cannot occur because there are no
+deallocations.
+
+In this approach the following constructs are not allowed:
+
+   * Allocators
+   * User-defined storage pools
+   * Unchecked Deallocations
+
+Aspects should be applied to all access types in this approach, specifying a
+value of zero for the storage size.  Although the restriction No_Allocators is
+present, such clauses may be necessary to prevent any default storage pools
+from being allocated for the access types, even though the pools would never be
+used. A direct way to accomplish this is to use pragma Default_Storage_Pool
+with a parameter of :ada:`null` like so:
+
+   :ada:`pragma Default_Storage_Pool (null);`
+
+The above would also ensure no allocations can occur with access types that
+have the default pool as their associated storage pool (per
+:arm:`13.11.3\\(6.1\\/3\\) Default Storage Pools <13-11>`
+
```

Comment:
missed ")"
also the link renders as below (note back slash and forward slash), please ignore this comment if this is intended

![Screenshot from 2023-11-16 16-17-06](https://github.com/AdaCore/learn/assets/22798363/e8fbe16f-4e21-40d3-bbf5-a503de417e47)